plugins/woocommerce/src/Internal/Admin/BlockTemplates/BlockContainerTrait.php

```diff
@@ -44,6 +45,91 @@ protected function &add_inner_block( BlockInterface $block ): BlockInterface {
 
 	// phpcs:enable Squiz.Commenting.FunctionCommentThrowTag.WrongNumber
 
+	/**
+	 * Checks if a block is a descendant of the block container.
+	 *
+	 * @param BlockInterface $block The block.
+	 */
+	private function is_block_descendant( BlockInterface $block ): bool {
+		$parent = $block->get_parent();
+
+		if ( $parent === $this ) {
+			return true;
+		}
+
+		if ( ! $parent instanceof BlockInterface ) {
+			return false;
+		}
+
+		return $this->is_block_descendant( $parent );
+	}
+
+	/**
+	 * Remove a block from the block container.
+	 *
+	 * @param string $block_id The block ID.
+	 *
+	 * @throws \UnexpectedValueException If the block container is not an ancestor of the block.
+	 */
+	public function remove_block( string $block_id ) {
+		$root_template = $this->get_root_template();
+
+		$block = $root_template->get_block( $block_id );
+
+		if ( ! $block ) {
+			return;
+		}
+
+		if ( ! $this->is_block_descendant( $block ) ) {
+			throw new \UnexpectedValueException( 'The block container is not an ancestor of the block.' );
+		}
+
+		// If the block is a container, remove all of its blocks.
+		if ( $block instanceof ContainerInterface ) {
+			$block->remove_blocks();
+		}
+
+		// Remove block from root template's cache.
+		$root_template = $this->get_root_template();
+		$root_template->uncache_block( $block->get_id() );
+
+		$parent = $block->get_parent();
+		$parent->remove_inner_block( $block );
+
+		// Detach block from parent and root template.
+		$block->detach();
+
+	}
+
+	/**
+	 * Remove all blocks from the block container.
+	 */
+	public function remove_blocks() {
```

Comment:
I would almost suggest to change this to `remove_all_blocks`, `remove_blocks` makes me think I can pass an array of id's in, and it will remove all of the specific one's.

Reply:
I could go either way... I like the brevity of `remove_blocks`.